### DISCUSSION TITLE
Some Simulation Results with Wideband Dielectric models

### DISCUSSION THREAD

**Discussion by toammann** [thliebig/openEMS-Project#157] [General]
Hello,
in the last weeks I took a closer look at substrate losses. More specifically the Debye and Djordjevic-Sarkar [1] models. I think this is a good point to share the results I have so far.

The Djordjevic-Sarkar model was suggested by @biergaizi in the context of my [Microstrip to SMA transition model](https://github.com/thliebig/openEMS-Project/discussions/88). @VolkerMuehlhaus has opened a discussion thread for [wideband dielectric models](https://github.com/thliebig/openEMS-Project/discussions/99) where he published some simulations in Sonnet for comparison to openEMS. 

The Djordjevic-Sarkar was originally developed for FR4-type materials based on measured data. The idea is to have epsilon'' (imaginary part) approximately constant between two corner frequencies $\omega_1 = 10^{m_1}$ (lower limit) and  $\omega_2  = 10^{m_2}$ (upper limit). With this assumption it is possible to create a wide band model with only one epsilonR, tand pair at a single frequency. The model is implemented in several commercial simulators (CST, HFSS, Simberian). According to multiple sources this model works well for materials with an almost constant loss tangent (very common for PCB materials). The Djordjevic model is roughly speaking a Debye model with an infinite number of terms, but it can be approximated by a finite number of poles. With some math it is possible to solve for the unknowns by choosing an appropriate number of Terms. In a second step the Debye model can be (approximately) converted into the openEMS Lorentz model as defined in openEMS

## Motivation

Currently I am working on a openSource tunable band pass filter. The plan is to make a co-simulation of QUCS (models for lumped elements and the varactor) and openEMS (for the passive filter structure). To achieve a close agreement between measured and simulated results it is  critical to get all losses right. In the passive structures as well as in the lumped component models. This is why I started to look into the substrate losses.

## The DUT (Device Under Test)

The low cost Rosenberger 32K242-40ML5 connector combined with my wide band transition is an enabler for accurate de-embedding using the [IEEEP370 open source 2xThrough](https://opensource.ieee.org/elec-char/ieee-370) method. Accurate wide band de-embedding can be achieved. From the two projects I have two calibration lines of different lengths (See Figure below). By de-embedding the half sections of the short transmission line from  the longer one connector effects are removed. The length difference is 23.5mm.

![dut_2xthru](https://github.com/thliebig/openEMS-Project/assets/9382233/dd68c6c2-0f8e-4c35-9c5e-99132123cf79)

 Both are fabricated on the [Aisler 6Layer HD stackup](https://community.aisler.net/t/6-layer-hd-pcb-stackup/1703). To get a reasonable substrate thickness for RF (filter) design the upper prepreg and core have to be combined. This is bad for the insertion loss due to the lossy core martial. However, this is the price to pay when using an ultra low cost PCB.

The material parameters are taken from the [substrate datasheets (min./max values)](https://community.aisler.net/uploads/short-url/1oBcIOZ5qPiZRUQdlzBzAUm1VJg.pdf) linked by Aisler and the [Panasonic datasheet](https://industrial.panasonic.com/content/data/EM/PDF/CDS_HF_R-1566(W_WN)_R-1551(W_WN)_22070140.pdf).

The maximum value for the laminate is a very high tand=0.035 value. Typical values in the datasheet are around 0.013 (a very low value for an FR4 laminate). These datasheet values are known to be very optimistic. Are these values trustworthy? The span between typical and max. is a too high in my opinion. Therefore, I went with a typical value for FR4 of tand=0.02. For the prepreg the datasheet typical value form the datasheet was taken. 

```
Prepreg_1080_Panasonic-R-1551W, Er=4.3,tand = 0.013
Core_7628_Panasonic-R-1566W, Er = 4.6, tand=0.02
```

The figure below shows a de-embedded VNA measurement of the transmission line.

![meas_deemb](https://github.com/thliebig/openEMS-Project/assets/9382233/4d1ccb90-7339-44dd-8833-36225d114429)

## OpenEMS Simulation Model

A openEMS simulation model was written to recreate the DUT from above with a length of 23.5mm. 

Four test cases were simulated
- Lossless Simulation (no substrate and metal losses)
- Conductor losses only (AddConductingSheet())
- Substrate losses only (AddDjordjevicSarkarMaterial())
- Combined losses (AddConductingSheet() and AddDjordjevicSarkarMaterial())

![sim_model](https://github.com/thliebig/openEMS-Project/assets/9382233/51e8d9e9-9860-42e5-b8f4-e1c383338d48)

The current result of my work is a function `AddDjordjevicSarkarMaterial()` that calculates a multi term wide band material model. In short - without providing too much details - the following steps are performed:

1. Calculate Djordjevic-Sarkar model from the users $\epsilon_r$ and $tan_\delta$
2. Select suitable frequency points to perform a fit to a multi-term Debye Model
3. Fit to multi-term Debye Model
4. Abuse the Lorentz model implementation to reassemble the Debye model

Example call:
```
    CSX = AddDjordjevicSarkarMaterial(CSX, 'Sub01', 4.3, 0.015, 6e9, 200e9, 'f1', 10^4/(2*pi));
```

Here are debug plots that show the model fit of  multi term Debye and Lortentz model in linear and logarithmic scale

![calc_model_log](https://github.com/thliebig/openEMS-Project/assets/9382233/d61e92fb-b866-435d-a35d-8e77b05d44d6)
![calc_model_lin](https://github.com/thliebig/openEMS-Project/assets/9382233/ee1e73a6-2981-4787-9603-f1375b155349)

The "abuse" of the Lorentz model comes with some [stability problems](https://github.com/thliebig/openEMS-Project/discussions/140). In the time domain the frequency dependent material properties are handled by an equivalent circuit approach. My function basically forces the equivalent circuit of the Lorentz model to be equal to the Debye one. Maybe here some instability occurs.

In its current state I would not consider `AddDjordjevicSarkarMaterial()` to be save to use. The cleanest way would be to add the Debye model into openEMS along with the Lorentz model (This is certainly a topic of discussion). However, I consider this is a long term project.  If anybody needs a wide band dielectric model desperately (like me :).:`AddDjordjevicSarkarMaterial()` can be used at own risk.

## Results

The simulation model's outputs for the four test cases can be plotted on a single graph, following the standard practice for these types of comparison. Transmission losses in the lossless case are almost zero. I noticed that at these low attenuation values even the PML boundary thickness has an influence.

![meas_vs_sim_s21](https://github.com/thliebig/openEMS-Project/assets/9382233/080a289c-2260-4b67-b3d6-e692ad746ee5)

The substrate and conductor losses were added sequentially. For comparison the measured transmission is also included. We can see, that the simulation has too low losses. Even when  conductor losses are considered by `AddCondSheet()` and a wide band model for the substrates. I think that the conductor losses are probably not treated well. Copper roughness is not included. This is in my experience required. What do you think?

It would be nice to have a simulation of a commercial simulator for comparison. The results of the Djordjevic-Sarkar model agree with Sonnet ([see here](https://github.com/thliebig/openEMS-Project/discussions/99)]. But this simulation is without conductor losses.

For completeness here is the plot for S11.

![meas_vs_sim_s11](https://github.com/thliebig/openEMS-Project/assets/9382233/57ee77de-733c-4cc0-a2b9-339ad16fb618)

## Problems

- For my filter PCB I cannot use the `AddCondSheet()` model. The filter is highly sensitive to coupling effects between transmission lines. I must model the traces with their proper thickness. To my knowledge there is no way to treat metal losses of volumetric shapes without a very fine mesh to capture the skin effekt?
- Stablility Problems due to Lorentz model approximation
- Losses still not treated correctly...
- Unfortunately, I have no comparison to a commercial simulator. 

## References
[1] Djordjevic, Antonije R., et al. "Wideband frequency-domain characterization of FR-4 and time-domain causality." IEEE Transactions on electromagnetic compatibility 43.4 (2001): 662-667.

## Link to model
Please find the code and measured touchstone files in my repository:
https://github.com/toammann/OpenEMS_MultiTermDebyeExperiments

Any comments or suggestions are welcome.

Regards,
Tobias

**Comment by biergaizi** [comment]
I'm glad to see that my original one-post suggestion of using "Djordjevic-Sarkar model with numerical curve fitting into the Lorentz model" has not only been implemented in code but is being verified experimentally, with the limitations highlighted. The progress is truly impressive. Thanks for your contribution to the entire community.

If I read this report correctly, preliminary results are:

1. Wideband dielectric loss is successfully modeled, both in terms of Djordjevic-Sarkar model itself, and in openEMS simulations, the results agree with SONNET.
2. There are numerical instabilities, as Djordjevic-Sarkar model is designed to use with Debye material model, but openEMS uses Lorentz materials, and fitting Debye's parameters into Lorentz is an imperfect process. In the future, we probably need a native Debye model as well.
3. Wideband conductor loss is not modeled correctly, as the Djordjevic-Sarkar model is only a dielectric material model. One still needs to develop a realistic conductor loss model including the effects of surface roughness, future works are needed to solve this problem to enable very wideband circuit board modeling in openEMS.

I'd like to add another separate point that was previously reported to me:

* Simulation performance is often extremely low when openEMS's Lorentz material is turned on, the speed can degrade to 1% to 10% of the original performance. I'll investigate and possibly fix this problem when I have time (right now I'm still focused on optimizing the main engine).

**Comment by VolkerMuehlhaus** [comment]
Impressive work on wideband dielectric loss and excellent presentation of results, Tobias, congratulations!!!

I want to comment on one error source for **conductor loss** when simulating it as **zero thickness sheet** in EM. Conductor loss for thin sheet model uses a frequency dependent surface impedance calculated from effective cross section in the skin effect regime. 

However, in the real thick conductor current can flow on the top side, or bottom side, or a mixture of both. This makes it difficult to estimate the effective current cross section in skin regime.

![current_top_bot](https://github.com/thliebig/openEMS-Project/assets/47922981/a183e163-bb26-4130-ac9f-c74636ca47d4)

For a stripline with grounds above and below, we have a symmetric case with equal "skin sheets" on top and bottom, resulting in twice the skin thickness for the thin sheet model. For microstrip, most current is on the bottom of the conductor only, so that the resulting thin sheet model is more accurate when using only one skin depth for the loss calculation. 

The difficulty here is that the thin sheet **EM solution** does not solve for top and bottom side separately, so this "current ratio" is only an estimate. For thin sheet conductors in the Sonnet EM solver, this current ratio is actually a user defined parameter, with default set to the "pessimistic" case of single sided current flow. For OpenEMS I don't know what the underlying calculation assumes, one or two skin depths.

I wanted to simulate your testcase in Sonnet today, to compare the optimistic (2 skin sheets) and pessimistic (1 skin sheet) case, but my license expired a few days ago and I need to wait for license renewal. Maybe we can then add a separate topic on conductor loss modelling, so that we don't disctract from your great progress in dielectric loss modelling presented here!

**Reply by biergaizi** [reply]
> For OpenEMS I don't know what the underlying calculation assumes, one or two skin depths.

openEMS's lossy conducting sheet model is basically just a Lorentz material with 30-pole parameters, defined in the following code:

```
unsigned int numOptPara=30;
double omega_stop[30]={20,30,45,67.5,101.25,151.875,227.8125,341.71875,512.578125,768.8671875,1153.300781,1729.951172,2594.926758,3892.390137,5838.585205,8757.877808,13136.81671,19705.22507,29557.8376,44336.7564,66505.1346,99757.7019,149636.5529,224454.8293,336682.2439,505023.3659,757535.0488,1136302.573,1704453.86,2556680.79};
double omega_critical_5[30]={0,0,0,1.566,6.8445,10.4085,15.1419375,22.30284375,33.45426562,50.28391406,75.42587109,113.1388066,169.70821,254.5623149,381.8434724,572.7652086,859.1478129,1288.721719,1933.082579,2899.623869,4349.435803,6524.153704,9786.230557,14679.34583,22019.01875,33028.52813,35503.14262,119084.5097,180444.8486,704280.3349};
double omega_critical_20[30]={0,0,0,0,1.41075,2.7135,4.100625,5.90034375,8.6113125,12.86571094,19.37545312,29.06317969,43.59476953,65.3921543,98.08823145,147.1323472,220.6985208,331.0477811,496.5716717,744.8575075,1117.286261,1675.929392,2513.894088,3770.841132,5656.261698,8484.392547,10504.48601,11363.02573,11135.76522,7499.596983};
double g[30]={0.1370843401,0.1244327709,0.1087425925,0.09215450299,0.07639249525,0.06239206861,0.05068701937,0.04124161681,0.03364925947,0.0274825647,0.02244156564,0.01832317373,0.01496080786,0.01221545053,0.009973874666,0.008143635713,0.006649251918,0.005429092739,0.00443283702,0.003619397792,0.002955227653,0.002412935361,0.001970156167,0.001608629812,0.001313423397,0.001072422468,0.0009031383439,0.0007674973627,0.0006482579571,0.0005450699565};
double r1[30]={1.302754546,1.2992612,1.320033314,1.41829213,1.674326257,2.138770494,2.769126112,3.498802047,4.311446869,5.266204065,6.443980095,7.893402541,9.667414736,11.84010502,14.50110421,17.76014698,21.75163911,26.64019322,32.6274156,39.96021835,48.94100192,59.94011376,73.41109983,89.90930992,110.1195785,134.8631576,149.2770459,153.5258961,148.9373215,119.3522908};
double l1[30]={0.9542567116,0.9412051444,0.8999655165,0.8189903271,0.702028543,0.5718817624,0.4577262324,0.3689027719,0.3004106661,0.2455514861,0.2005633225,0.1637496639,0.1337010005,0.1091664663,0.0891340699,0.07277768839,0.05942276018,0.04851850861,0.03961522618,0.03234573021,0.02641021614,0.02156389712,0.0176069076,0.01437606898,0.0117376269,0.009584103516,0.008750848841,0.008547706786,0.00854127488,0.008801399365};
double r2[30]={10.62245057,10.21952962,10.24755832,10.99147713,12.70450822,15.59830718,19.55714986,24.32580904,29.88733852,36.55680626,44.75374078,54.81575133,67.13533424,82.22360504,100.7029064,123.3353165,151.0542147,185.0027444,226.5809553,277.5035137,339.8704104,416.2534638,509.8022059,624.3729288,764.7279941,936.5525496,1000,999.9999981,999.999984,999.9999985};
double l2[30]={0.7925524826,0.6926195458,0.5814718257,0.4784644339,0.3916667501,0.3204661897,0.2620413998,0.2140939121,0.1748272154,0.1427375388,0.1165424396,0.09515681664,0.07769522035,0.06343788358,0.05179681795,0.04229192785,0.03453121766,0.02819462457,0.0230208182,0.01879642336,0.01534721999,0.01253095815,0.01023149158,0.008353988465,0.006820957973,0.005569332359,0.004553963932,0.003696380632,0.002978504727,0.002391295771};
```

These material parameters are generated by the script `OptimizeCondSheetParameter.m`, the source code can be read here: https://github.com/thliebig/openEMS/blob/5f36e7f3a2367123f00999491a069aed50c6f244/FDTD/extensions/OptimizeCondSheetParameter.m

The model appears to be a rather basic one, defined mainly by these equations:

```
    Ys = tanh((1+1j)*sqrt(Omega_fine))/(1+1j)./sqrt(Omega_fine);

    err = (CalcYDiff(Omega_fine,X))./real(1./Ys);
    
    fc = Omega_fine(find(abs(err)>0.2,1,'last'));
    if (isempty(fc))
        fc = 0;
    end

function Ydiff =  CalcYDiff(omega, X)
     Ys = tanh((1+1j)*sqrt(omega))/(1+1j)./sqrt(omega);
     Y = X(1) + 1./(X(2)+1j*omega*X(3)) + 1./(X(4)+1j*omega*X(5));
     Ydiff = real(1./Ys)-real(1./Y);
```

Before they're used as input to the underlying Lorentz material code, these Lorentz material parameters are further modified by a scale factor calculated according to simulation conditions such as conductivity and thickness in [operator_ext_conductingsheet.cpp](https://github.com/thliebig/openEMS/blob/5f36e7f3a2367123f00999491a069aed50c6f244/FDTD/extensions/operator_ext_conductingsheet.cpp#L206).

It should not be difficult to check the underlying assumptions of the current model by reviewing the code involved. It should also be possible to create a better lossy conductor model without major changes to the engine if it can be expressed in terms of a set of Drude/Lorentz/Debye material parameters.

**Reply by toammann** [reply]
Here is a small addition to @biergaizi's information. In [FDTD/extensions/operator_ext_conductingsheet.h](https://github.com/thliebig/openEMS/blob/71dde7ea49918c033ae91487c38db959d381736f/FDTD/extensions/operator_ext_conductingsheet.h#L30) is a link to the paper used for the  conducting sheet model.

```
/*!
  FDTD extension for a conducting sheet model as described in:
  Lauer, A.; Wolff, I.; , "A conducting sheet model for efficient wide band FDTD analysis of planar waveguides and circuits," Microwave Symposium Digest, 1999 IEEE MTT-S International , vol.4, no., pp.1589-1592 vol.4, 1999
  doi: 10.1109/MWSYM.1999.780262
  URL: http://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=780262&isnumber=16934
  */
```

**Reply by biergaizi** [reply]
> In [FDTD/extensions/operator_ext_conductingsheet.h](https://github.com/thliebig/openEMS/blob/71dde7ea49918c033ae91487c38db959d381736f/FDTD/extensions/operator_ext_conductingsheet.h#L30) is a link to the paper

Thanks for pointing it out! I've completely missed this comment in spite of working with the codebase for several months...

**Comment by biergaizi** [comment]
> We can see, that the simulation has too low losses. Even when conductor losses are considered by AddCondSheet() and a wide band model for the substrates. I think that the conductor losses are probably not treated well. Copper roughness is not included. This is in my experience required. What do you think?

Can this disparity be entirely attributed to surface roughness, or are we also missing something else? Have you tried to check the agreement between openEMS with another model (simple equations, analytical models, 2D models like Si9000, or another full-wave model) by modeling the same fixture, but with the copper's surface roughness turned off?

**Reply by VolkerMuehlhaus** [reply]
In the meantime I simulated Tobias model in a commercial FDTD solver. I used the STL data exported from AppCSXCAD and thin sheet metal model, with loss calculation set to wideband lossy for metal and dielectric, no conductor roughness. 

Result for insertion loss agrees very well with openEMS calculation shown by Tobias, I get -1.3dB @ 20 GHz. My conclusion is that Tobias did an excellent job with his math and model, and the extra loss in measurement is indeed some additional effect not included in the model.

**Reply by toammann** [reply]
> Can this disparity be entirely attributed to surface roughness, or are we also missing something else? 

I was hoping that someone with access to commercial software can verify the results. Awesome that Volker was able to do so!

 Unfortunately, the tolerance range for tand is quite high for the FR4 material used. It may be that the real tand value is slightly higer. So, some uncertainty exists in this parameter.

>  Have you tried to check the agreement between openEMS with another model (simple equations, analytical models, 2D models like Si9000, or another full-wave model) 

No, I have not done this yet. I will request further information from Aisler maybe they can provide some information about the used copper foil to get data for copper roughness. A simulation with metal losses in a commercial software would also be appreciated (Does Sonnet have copper roughness models?).

Thanks again,  for the time you spent in the forum. I have learned quite a lot here!

Regards,
Tobias

**Reply by VolkerMuehlhaus** [reply]
> A simulation with metal losses in a commercial software would also be appreciated (Does Sonnet have copper roughness models?).

I could not check in Sonnet yet (waiting for license file), but verified results including surface roughness using two other commercial solvers, one FDTD and one Method of Moments solver.

Insertion loss **without** roughness is 1.3dB at 20 GHz (FDTD) and 1.35 dB (MoM), using the same assumptions that you used for your model (thin sheet metal model). This confirms your model and results.

When I include a (large!) roughness value of **6um** using the **Hammerstad-Jensen model**, the additional insertion loss is **0.1dB** in both solvers.
The MoM solver offers different roughness models, and according to the manual the **multi-level hemispherical model** is more accurate than Hammerstad-Jensen. Using that hemispherical model, insertion loss increases by **0.2dB** for 6um roughness.

Not sure what a realistic roughness value is, but I guess my 6um is already pessimistic.

[1] Hammerstad and Jensen, “Accurate models for microstrip computer aided design,” in IEEE MTT-S Int. Microw. Symp. Dig., pp. 407–409, May 1980

**Reply by VolkerMuehlhaus** [reply]
Update: Using the metal roughness modell in Sonnet, the effect is much larger than seen with the other solvers: S21 = -2.1dB @ 20 GHz with 6um roughness, -1.4dB @ 20 GHz without roughness. 

I can't explain why that is so much larger effect compared to the other solvers. In the [manual ](https://www.sonnetsoftware.com/support/help-18/users_guide/Sonnet%20User's%20Guide.html?SonnetUsersGuide.html)they link this reference:

> Allen F. Horn, John W. Reynolds, Patricia A. LaFrance, James C. Rautio, "Effect of conductor profile on the insertion loss, phase constant, and dispersion in thin high frequency transmission lines," DesignCon Best Paper Award, February 2010.

Note that the 6um roughness value in my tests is just a guess, most likely a very pessimistic guess. Tobias, could you find out what roughness Aisler specifies?

**Reply by toammann** [reply]
Hey Volker,
Thanks for doing the verification simulations. I have contacted Aisler. Hopefully, they can provide some data. 

The used stack up is - most certainly - not optimized for RF. It was, for me, just very cost-effective to source. The  copper roughness value may indeed be high value. In PCB manufacturing, high surface roughness on the dielectric side is preferred to strengthen the bond between the copper foil and the dielectric. It increases the immunity against delamination. Aisler had most certainly no RF applications in mind and optimized for durability.  

A Google search returned 6um for "standard copper foil". However, the source does not look trustworthy to me :). I did even ask  ChatGPT :)

```Sure, a typical value for the surface roughness of a non-RF optimized copper foil used in PCB design might be around 5 to 10 micrometers (µm) root mean square (RMS). This can vary slightly based on the specific manufacturer and the production processes involved.```

> I can't explain why that is so much larger effect compared to the other solvers. 

I have noticed that models for conductor loss can produce significantly different outputs. Especially, the Hammerstad-Jensen model has some terms in it that saturate at higher frequencies. This prevents the loss term from growing. However, the loss in real models does indeed grow. The Hammerstad-Jensen model underestimates' conductor loss for higher frequencies significantly. I think the CST help recommends this model only up to 5 GHz, or so? I have to check..

Getting the metal losses right will be another hard to solve task...

## Verify using WCALC
I have now verified the results above using an equation-based microstirp analysis tool. WCALC is a great tool. It provides references to all the equations used. It is also free. It is supplied as a web calculator and also in binary form to be installed on a computer. 

https://wcalc.sourceforge.net/about.html

It uses the Hammerstad-Jensen model where the saturation effect can be observed. The loss value does not change much for higher frequencies and higher roughness values.

### Substrate losses only
![image](https://github.com/thliebig/openEMS-Project/assets/9382233/f59a2349-d778-494e-a48e-b7fc57532ff1)

The two dielectrics I have simulated above have different values of tand (I guessed the core metal to have a loss tangent around 0.02). I went simply with the mean value of the two tand=0,0165. As the core is thicker, I think the effective loss tangent will be slightly higher. Wcalc reports a loss value of 1.18dB for the substrate losses only case. This is very close to my simulated 1.22dB loss. I think the substrate losses are treated correctly

### Substrate and metal losses (Hammerstad-Jensen with saturation effect)

![image](https://github.com/thliebig/openEMS-Project/assets/9382233/a93e58ec-2072-414a-b709-4b0e751fb964)

The metal losses do not change the result significantly. Here is a link to the used models of WCALC.
https://wcalc.sourceforge.net/microstrip.html

Maybe it is a good idea to open a new discussion thread for metal losses to separate this discussion from the substate losses, as suggested by Volker earlier.

Regards!

**Comment by biergaizi** [comment]
I just noticed the PCB has gold plating. Thus, another source of error is the losses from gold plating. Losses from the ENIG surface finish has a magnitude comparable to surface roughness due to the use of nickel and gold.

Let's continue this discussion at https://github.com/thliebig/openEMS-Project/discussions/161#discussioncomment-7836122 instead.

**Comment by biergaizi** [comment]
I'm now trying to model this PCB using Si9000 for comparison. But I have find it's difficult to understand the PCB layout and what exactly are simulated and measured. Could you please clarify these details: Which layer is the signal layer? Which layer is the ground layer? How long is the line (after de-embedding)? What happened to the unused layers (Which layer is removed? Which layer is via-stiched to the ground layer)?

Since you've mentioned that both prepreg and core are included in the dielectric, my guess is that the signal layer is layer 1, the ground layer is layer 3, meanwhile the layer 2 copper under the trace is removed, and all the other unused layer in via-stitched to layer 2? Thus, this test fixture can be modeled as a PCB with two metal layers and two dielectric layers? Is that correct?

**Reply by VolkerMuehlhaus** [reply]
Line length to be simulated is 23.5mm, width = 0.65mm, gap from line to side ground = 1.225mm

**Signal + side grounds on layer 1**
_Core t=0.2mm er=4.6 tand=0.02_
**side grounds only on layer 2**
_Prepreg t=0.136mm er=4.3 tand=0.013_
**solid ground on layer 3**

The vias connect side grounds to the bottom ground, diameter 0.3mm, distance from side ground edge to via center 0.3mm

**Comment by biergaizi** [comment]
As proposed, I've opened a new thread: [Conductor loss for wideband models](https://github.com/thliebig/openEMS-Project/discussions/161) to separate further discussion about conductor loss modeling from dielectric modeling.

**Comment by biergaizi** [comment]
## Comparison with Si9000

Polar Instrument's Si9000 is a transmission line calculator widely used by printed circuit board manufacturer. It's primarily an impedance calculator based on a 2D MoM electromagnetic field solver, but new versions also support frequency-dependent analysis, including S-parameter analysis, substrate εr extrapolation via Svensson-Djordjevic model, and surface roughness modeling using Hammerstad, Groisse, Huray, and Cannonball-Huray models.

### Dielectric Loss Only

Here's a comparison of openEMS (Djordjevic) and Si9000's dielectric loss calculation.

To eliminate conductor loss, an extremely high, unphysically value is used as the metal conductivity in Si9000. To extrapolate dielectric's εr, the prepreg and core meterial's dielectric constant and loss tangent are specified at and only at 6 GHz. Internally, Si9000 uses the Svensson-Djordjevic model to approximate the wideband material property. This is similar to the Djordjevic-Sarkar model used in the original openEMS simulation.

As one can see, both models are indistinguishable. 

![Comparison with Si9000 and openEMS with PEC Trace](https://github.com/thliebig/openEMS-Project/assets/1310693/ba57f87c-cc2b-4e57-ae57-fd3dd21196f9)

### Dielectric Loss + Smooth Metal Loss

The following is a comparison of openEMS (Djordjevic + CondSheet) with Si9000, with both dielectric loss and smooth metal loss. As one can see, the agreement is still respectable to 0.1 dB at 20 GHz.

![Comparison with Si9000 and openEMS with Smooth Trace](https://github.com/thliebig/openEMS-Project/assets/1310693/fdda56c9-caef-48a9-b448-dc295d808756)

### Rough Surface

Next, two different surface roughness models - Hammerstad and Cannonball-Huray - are compared with the results from smooth conductor. In this comparison, the worst preset value from Si9000 is used. In the Hammerstad model, the roughness on both sides of the copper foil is set to Rq = 8.25 µm RMS. In the Cannonball-Huray model, the roughness on both sides is set to Rz = 7.5 µm.

Since I've already explained the assumption, properties and application of each model and parameter in a separate thread, [*Conductor loss for wideband models*](https://github.com/thliebig/openEMS-Project/discussions/161), so it will not be repeated here.

![Comparison in Si9000 with Smooth and Rough Traces](https://github.com/thliebig/openEMS-Project/assets/1310693/62199322-e526-415b-abef-727617ef8126)

As one can see, Cannonball-Huray predicts a worse attenuation than Hammerstad, as expected, since it's widely accepted that Hammerstad underestimates losses (but do note that these two models use different paramaters so they're technically not directly comparable). However, even with the assumption of Rz = 7.5 µm, a pretty bad roughness value, the simulated result is still 0.3 dB away from measurement.

Thus, there are three possible explanations for this disparity:

1. Copper foil is really bad and rougher than Rz = 7.5 µm.
2. ENIG-plate traces is an addition contribution to conductor loss, which is neglected here.
3. Errors in VNA measurement, especially the de-embedding process. 

I suspect it's a combination of these three factors. 

### Input Parameters Used for Calculation

#### Lossless Calculation

|        Parameter             |      Value      |
|------------------------------|-----------------|
| Parameter Entry Units        |  Millimeters    |
| Interface                    |  Standard       |
| G.S. Convergence             |  Fine (Slower)  |
| Substrate 1 Height (H1)      |  0.2 mm         |
| Substrate 1 Dielectric (Er1) |  4.6            |
| Substrate 2 Height (H2)      |  0.138 mm       |
| Substrate 2 Dielectric (Er2) |  4.3            |
| Upper Trace Width (W1)       |  0.64 mm        |
| Lower Trace Width (W2)       |  0.65 mm        |
| Trace Thickness (T1)         |  0.018 mm       |

* Impedance (lossless): 48.94 Ω

#### Frequency-Dependent Calculation

|        Parameter             |      Value      |
|------------------------------|-----------------|
| Length of Line (LL)          | 23.5 mm         |
| Trace Conductivity (TC)      | 5.80e7 S/m      |
| Loss Tangent                 | 0.020           |
| Rise Time (ps)               | 10 ps           |
| Frequency Minimum (Fmin)     | 1000 MHz        |
| Frequency Maximum (Fmax)     | 20 GHz          |
| Frequency Steps              | 200             |
| Frequency Distribution       | Logarithmic     |
| Result Presentation          | Length of Line  |
| Use Extended Substrate Data  | Yes             |
| Substrate 1                  | Causal Extrapolate / Core 7628       |
| Substrate 2                  | Causal Extrapolate / Prepreg 1080    |

#### Causally Extrapolate Substrate Data (Svensson-Djordjevic)

| Substrate | Frequency |  Er   |   TanD   |
|---------- |-----------|-------|----------|
|     1     |  6.00e9   |  4.6  |   0.020  |
|     2     |  6.00e9   |  4.3  |   0.013  |

**Comment by biergaizi** [comment]
I must point out an important error that we've previously missed. The finite n-term Debye model is in fact implemented in openEMS. There should be no need to abuse the Drude/Lorentz model. I wrote a programming and math note as a draft in my fork. 

See https://fasterems.github.io/openEMS-Project/concepts/dispersive.html

**Reply by toammann** [reply]
Hello,
unbelievable! I cannot believe that I missed this. The reason may be that it was not mentioned in the documentation. 

I don’t expect the results above to change because the Lorentz Debye fit was pretty good.

With the Debye model already implemented, using and adding the Djordjevic-Sarkar model becomes very easy. The two modes of using the Debye model are then:

1. Provide a list of Er, tand pairs over frequency and use the Debye Model standalone (each pair generates one pole; typically one pole per decade is required). For convenience a wrapper function which handles the tand to conductivity conversion could be beneficial. 

2. Provide a single Er, tand pair at one frequeny and use the wideband Djordjevic-Sarkar model which internally uses the Debye model

I think the functions should be updated and tested. Thanks for the hint to the Debye model!

Regards Tobias

**Reply by toammann** [reply]
Hello,

I finally had time to test the Debye model, and it performs as expected, producing similar results.

With the stability issues from the unnecessary Lorentz fit out of the picture, I believe the fitting function is now ready to be merged. I’ll submit a pull request shortly. 

The function is now much easier. The same fitting approach could be used to create a fitting function for a wideband model from a list of (frequency, ε_r, tanδ) triplets.

I also plan to contribute documentation updates on this topic.

@biergaizi — I noticed a minor error in your documentation: the correct keyword for `Epsilon_Infinity `is `Epsilon`, not `eps_r`.
![grafik](https://github.com/user-attachments/assets/f2ea2b7e-d32d-4c5d-8a3c-6695b9de33a2)

**Reply by biergaizi** [reply]
@toammann Typo fixed. Note that the URL has changed, it's now called `dispersive.html`. Please update your bookmark if you had the old URL.

**Reply by toammann** [reply]
Pull Request is open:

https://github.com/thliebig/CSXCAD/pull/64

Regards,
Tobias
